packages/cli/test/integration/passwordReset.endpoints.test.ts

```diff
@@ -0,0 +1,231 @@
+		expect(response.statusCode).toBe(400);
+	}
+});
+
+test('POST /forgot-password should fail if user is not found', async () => {
+	const owner = await Db.collections.User!.findOneOrFail();
+	const authOwnerAgent = await utils.createAuthAgent(app, owner);
+
+	config.set('userManagement.emails.mode', 'smtp');
+
+	const response = await authOwnerAgent
+		.post('/forgot-password')
+		.send({ email: utils.randomEmail() });
+
+	expect(response.statusCode).toBe(404);
+});
+
+test('GET /resolve-password-token should succeed with valid inputs', async () => {
+	const owner = await Db.collections.User!.findOneOrFail();
+	const authOwnerAgent = await utils.createAuthAgent(app, owner);
+
+	const resetPasswordToken = uuid();
+
+	await Db.collections.User!.update(INITIAL_TEST_USER.id, { resetPasswordToken });
+
+	const response = await authOwnerAgent
+		.get('/resolve-password-token')
+		.query({ userId: INITIAL_TEST_USER.id, token: resetPasswordToken });
+
+	expect(response.statusCode).toBe(200);
+});
+
+test('GET /resolve-password-token should fail with invalid inputs', async () => {
+	const owner = await Db.collections.User!.findOneOrFail();
+	const authOwnerAgent = await utils.createAuthAgent(app, owner);
+
+	config.set('userManagement.emails.mode', 'smtp');
+
+	const first = await authOwnerAgent.get('/resolve-password-token').query({ token: uuid() });
+
+	const second = await authOwnerAgent
+		.get('/resolve-password-token')
+		.query({ userId: INITIAL_TEST_USER.id });
+
+	for (const response of [first, second]) {
+		expect(response.statusCode).toBe(400);
+	}
+});
+
+test('GET /resolve-password-token should fail if user is not found', async () => {
+	const owner = await Db.collections.User!.findOneOrFail();
+	const authOwnerAgent = await utils.createAuthAgent(app, owner);
+
+	config.set('userManagement.emails.mode', 'smtp');
+
+	const response = await authOwnerAgent
+		.get('/resolve-password-token')
+		.query({ userId: INITIAL_TEST_USER.id, token: uuid() });
+
+	expect(response.statusCode).toBe(404);
+});
+
+test('POST /change-password should succeed with valid inputs', async () => {
+	const owner = await Db.collections.User!.findOneOrFail();
+	const authOwnerAgent = await utils.createAuthAgent(app, owner);
+
+	const resetPasswordToken = uuid();
+	await Db.collections.User!.update(INITIAL_TEST_USER.id, { resetPasswordToken });
+
+	const passwordToSet = utils.randomValidPassword();
+
+	const response = await authOwnerAgent.post('/change-password').send({
+		token: resetPasswordToken,
+		id: INITIAL_TEST_USER.id,
+		password: passwordToSet,
+	});
+
+	expect(response.statusCode).toBe(200);
+
+	const authToken = utils.getAuthToken(response);
+	expect(authToken).toBeDefined();
+
+	const { password: storedPassword } = await Db.collections.User!.findOneOrFail(
+		INITIAL_TEST_USER.id,
+	);
+
+	const comparisonResult = await compare(passwordToSet, storedPassword!);
+	expect(comparisonResult).toBe(true);
```

Comment:
we should also check that the password is not saved in clear-text
```suggestion
	expect(comparisonResult).toBe(true);
	expect(storedPassword).toNotEqual(passwordToSet);
```